front/lib/resources/data_source_view_resource.ts

```diff
@@ -138,7 +136,7 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
     auth: Authenticator,
     vault: VaultResource,
     dataSource: DataSourceResource,
-    parentsIn: string[] | null
+    parentsIn: string[]
```

Comment:
This is a major change, can you at least outlined in the PR's description? Will also push to have this in a dedicated PR.

Reply:
Adding it to the description

Reply:
IMO it's not necessary to have it in a separate PR. We want to make that users will not be able to select an entire connection. Having it here, makes sure we are aligned both in terms of UI and API

Reply:
I'm fine with both as long as it's explicit in the PR description, since reverting this PR will mean running a script to fix those views.